src/NetAnalyzers/UnitTests/Microsoft.NetCore.Analyzers/Runtime/DetectPreviewFeatureUnitTests.cs

```diff
@@ -0,0 +1,558 @@
+                AttributeTargets.Delegate |
+                AttributeTargets.Interface |
+                AttributeTargets.Enum |
+                AttributeTargets.Constructor |
+                AttributeTargets.Method |
+                AttributeTargets.Property |
+                AttributeTargets.Field |
+                AttributeTargets.Event, Inherited = false)]
+    public sealed class RequiresPreviewFeaturesAttribute : Attribute
+    {
+        public RequiresPreviewFeaturesAttribute() { }
+    }
+
+";
+
+        [Fact]
+        public async Task TestMethodInvocation_Simple()
+        {
+            string csInput = @" 
+using System;
+namespace Preview_Feature_Scratch
+{" + setupPreviewAttribute +
+@"
+
+    public class Program
+    {
+        [RequiresPreviewFeatures]
+        public virtual void PreviewMethod()
+        {
+
+        }
+
+        static void Main(string[] args)
+        {
+            var prog = new Program();
+            [|prog.PreviewMethod()|];
+        }
+    }
+}";
+            await VerifyCS.VerifyAnalyzerAsync(csInput);
+
+        }
+
+        [Fact]
+        public async Task TestMethodInvocation_DeclareDerivedMethod()
+        {
+            string csInput = @" 
+using System;
+namespace Preview_Feature_Scratch
+{" + setupPreviewAttribute +
+@"
+
+    public class Program
+    {
+        [RequiresPreviewFeatures]
+        public virtual void PreviewMethod()
+        {
+
+        }
+
+        static void Main(string[] args)
+        {
+        }
+    }
+
+    public class Derived : Program
+    {
+        public Derived() : base()
+        {
+        }
+
+        public override void PreviewMethod()
+        {
+            [|base.PreviewMethod()|];
+        }
+    }
+}";
+            await VerifyCS.VerifyAnalyzerAsync(csInput);
+
+        }
+
+        [Fact]
+        public async Task TestMethodInvocation_DerivedNotMarked()
+        {
+            string csInput = @" 
+using System;
+namespace Preview_Feature_Scratch
+{" + setupPreviewAttribute +
+@"
+
+    public class Program
+    {
+        [RequiresPreviewFeatures]
+        public virtual void BaseMarked()
+        {
+
+        }
+
+        static void Main(string[] args)
+        {
+            var derived = new Derived();
+            derived.BaseMarked();
+        }
+    }
+
+    public class Derived : Program
+    {
+        public Derived() : base()
+        {
+        }
+
+        public override void BaseMarked()
+        {
+            [|[|base.BaseMarked()|]|];
+        }
+    }
+}";
+            await VerifyCS.VerifyAnalyzerAsync(csInput);
+
+        }
+
+        [Fact]
+        public async Task TestConstructor()
+        {
+            string csInput = @" 
+using System;
+namespace Preview_Feature_Scratch
+{" + setupPreviewAttribute +
+@"
+
+    class Program
+    {
+        [RequiresPreviewFeatures]
+        public Program()
+        {
+
+        }
+
+        static void Main(string[] args)
+        {
+            [|new Program()|];
```

Comment:
I'm having some trouble with the unit tests here. The analyzer seems to be invoked multiple times => multiple diagnostics are being triggered on the same line. I'm not sure if this is expected or if I'm doing something wrong?

cc @mavasani

Reply:
I believe it's not expected to have multiple diagnostics on the same line.

Reply:
@pgovind I ran the test locally, and I'm getting one diagnostic only.

Reply:
It was actually in multiple tests because I was using fields :) This is fixed now